doc/encoding.rdoc

```diff
@@ -132,7 +132,232 @@ returns the \Encoding of the concatenated string, or +nil+ if incompatible:
+
+The external encoding may be set by the open option +external_encoding+:
+
+  # File.
+  f = File.open('t.txt', external_encoding: 'ASCII-8BIT')
+  f.external_encoding                               # => #<Encoding:ASCII-8BIT>
+
+  # IO.
+  fd = IO.sysopen('t.txt')
+  io = IO.new(fd, external_encoding: 'ASCII-8BIT')
+  io.external_encoding # => #<Encoding:ASCII-8BIT>
+
+The external encoding may also set by method #set_encoding:
+
+  # File.
+  f = File.open('t.txt')
+  f.set_encoding('ASCII-8BIT')
+  f.external_encoding  # => #<Encoding:ASCII-8BIT>
+
+  # IO.
+  fd = IO.sysopen('t.txt')
+  io = IO.new(fd)
+  io.set_encoding('ASCII-8BIT')
+  io.external_encoding # => #<Encoding:ASCII-8BIT>
+
+==== Internal \Encoding
+
+If not +nil+, the internal encoding specifies that the characters read
+from the stream are to be converted to characters in the internal encoding;
+those characters become a string whose encoding is set to the internal encoding.
+
+The default internal encoding is +nil+ (no conversion):
+
+  # File.
+  f = File.open('t.txt')
+  f.internal_encoding # => nil
+
+  # IO.
+  fd = IO.sysopen('t.txt')
+  io = IO.new(fd)
+  io.internal_encoding # => nil
+
+The internal encoding may set by the open option +internal_encoding+:
+
+  # File.
+  f = File.open('t.txt', internal_encoding: 'ASCII-8BIT')
+  f.internal_encoding  # => #<Encoding:ASCII-8BIT>
+
+  # IO.
+  fd = IO.sysopen('t.txt')
+  io = IO.new(fd, internal_encoding: 'ASCII-8BIT')
+  io.internal_encoding # => #<Encoding:ASCII-8BIT>
+
+The internal encoding may also set by method #set_encoding:
+
+  # File.
+  f = File.open('t.txt')
+  f.set_encoding('UTF-8', 'ASCII-8BIT')
+  f.internal_encoding  # => #<Encoding:ASCII-8BIT>
+
+  # IO.
+  fd = IO.sysopen('t.txt')
+  io = IO.new(fd)
+  io.set_encoding('UTF-8', 'ASCII-8BIT')
+  io.internal_encoding # => #<Encoding:ASCII-8BIT>
+
+=== Script \Encoding
+
+A Ruby script has a script encoding, which may be retrieved by:
+
+  __ENCODING__ # => #<Encoding:UTF-8>
+
+The default script encoding is UTF-8;
+a Ruby source file may set its script encodingwith a magic comment
+on the first line of the file (or second line, if there is a shebang on the first).
+The comment must contain the word +coding+ or +encoding+,
+followed by a colon, space and the Encoding name or alias:
+
+  # encoding: ISO-8859-1
+  __ENCODING__ #=> #<Encoding:ISO-8859-1>
+
+=== Transcoding
+
+_Transcoding_ is the process of revising the content of a string or stream
+by changing its encoding.
+
+==== Transcoding a \String
+
+Each of these methods transcodes a string:
+
+String#encode :: Transcodes a string into a new string
+                 according to a given destination encoding,
+                 a given or default source encoding, and encoding options.
+
+String#encode! :: Like String#encode,
+                  but transcodes the string in place.
+
+String#b :: Transcodes a string into a new string whose encoding is ASCII-8BIT.
```

Comment:
I don't think `b` transcodes.  `b` never changes the contents of the string, it is similar to `dup.force_encoding('BINARY')`.

Reply:
Done.